### PR TITLE
Add release on tag

### PR DESCRIPTION
**Pull request by neubig**
This PR adds a github workflow to create a release when a new tag is created, it is copied directly from here: https://stackoverflow.com/a/75679739

Unfortunately, I'm not sure of any way to test whether this works other than incorporating this PR and creating a tag, then seeing if it works. If there are any other good ways to test this I'd be open to suggestions.

Fixes https://github.com/OpenDevin/OpenDevin/issues/1717
Fixes https://github.com/OpenDevin/OpenDevin/issues/1623

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Create release
+
+on:
+  push:
+    tags:
+      - v*
```
Comment (rbren) [inline .github/workflows/release.yml:6]: I think we removed the `v` prefix

Comment (rbren) [reply .github/workflows/release.yml:6]: You can check ghcr.yml for the syntax

**Comment by li-boxuan** [comment]
> Unfortunately, I'm not sure of any way to test whether this works other than incorporating this PR and creating a tag, then seeing if it works.

You could push the commits to your fork and create a tag to test it :)

**Comment by neubig** [comment]
Just noting that I'm testing this on my own fork.

Apologies in advance that I just realized that this resulted in a tag/release being made on the main repo because the workflow was not limited to "main":
https://github.com/OpenDevin/OpenDevin/releases/tag/1768%2Fmerge

I'll delete this.

**Comment by neubig** [comment]
Hmm, @rbren , I guess I was misunderstanding how tags/releases work. I thought that tags could be independent from releases, but I guess they're mostly the same thing? A release is basically a tag with additional info such as release notes?

Based on this, I don't think this PR makes all that much sense, so I'll close it, and also #1717.

#1623 is still somewhat valid though because our release notes are not very complete. As a solution to this, next time we cut a release, maybe we can use the `--generate-notes` option on the command line or through the web interface, which should automatically generate notes.

**Comment by li-boxuan** [comment]
On yes, on github a tag and a release are basically the same thing.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Shimada666** [comment]
I believe the release is meaningful because:

Releases can accommodate more content, allowing for a clear display of different versions, showcasing the features and fixes made in each version, and attributing the work to specific contributors (which can increase their sense of achievement). This is something that tags cannot achieve. 
image

Additionally, it looks better on the GitHub page, and having a clear release can make our product look more professional on the GitHub page.
image

Share some simple practice: using GitHub workflow to automatically generate a release from commit messages.
https://github.com/slidevjs/slidev/blob/f699eaf3ed0e557ab5032916376776e276ec8e3a/.github/workflows/release.yml#L1C1-L37C46
image


https://github.com/antfu/eslint-config/blob/bc891b56b5d7d7f435d97f8e042ea6b864ffd001/.github/workflows/release.yml#L1-L24

**Comment by neubig** [comment]
Yeah, I agree these are useful! We'll handle this in issue #1623

**Comment by Shimada666** [comment]
I think this PR is very meaningful! 
I support keeping the "v" prefix for tags, as this allows us to use the GitHub workflow to match specific prefixed tags and generate release information. Without the prefix, the workflow would automatically generate a release for every tag, which might not be what we want, as we might commit tags that are not version tags. 
Please proceed with it!

**Pull request closed without merging**